apis/envoy/v1alpha1/envoyconfig_types.go

```diff
@@ -182,6 +185,14 @@ type EnvoyConfig struct {
 	Status EnvoyConfigStatus `json:"status,omitempty"`
 }
 
+// GetEnvoyAPIVersion returns envoy's API version for the EnvoyConfigRevision
+func (ecr *EnvoyConfig) GetEnvoyAPIVersion() envoy.APIVersion {
+	if ecr.Spec.EnvoyAPI == nil {
+		return envoy.APIv2
+	}
+	return envoy.APIVersion(*ecr.Spec.EnvoyAPI)
+}
+
```

Comment:
To keep existent behavior, if the field is left unset it meand v2 is used. Going forward, when v2 is retired from the envoy codebase and v4 available, the default will be v3. We will keep 2 versions in the codebase at a given time, same as envoy does.